### ISSUE TITLE
Making assertions against indexers produces NS1004 (argument matcher used with a non-virtual member of a class) errors

### ISSUE THREAD

**Issue by simonmckenzie-oua** [nsubstitute/NSubstitute.Analyzers#190]
If I try to make an assertion against an indexer, I receive an NS1004 error - this seems to be a false positive.

Example:
```csharp
interface IInterfaceWithIndexer
{
    string this[int index] { get; set; }

    string SetSomething(int index, string value);
}

[Fact]
public void MockInterfaceWithIndexer()
{
    var mock = Substitute.For<IInterfaceWithIndexer>();

    mock.Received()[5] = Arg.Is("five");
    
    mock.Received().SetSomething(5, Arg.Is("five"));
}
```

![screenshot of NS1004 error in Visual Studio](https://user-images.githubusercontent.com/89581424/188355380-fd770683-8db6-47de-a3ea-5bcdc71ef758.png)

I _think_ I'm using the correct syntax for asserting against indexers - I used the guide here: https://nsubstitute.github.io/help/received-calls/#checking-calls-to-indexers

**Comment by simonmckenzie-oua** [comment]
Closed - I accidentally raised this with the wrong GitHub account 😳 
Replaced by #191
